dbt/adapters/spark/connections.py

```diff
@@ -26,6 +26,9 @@
 from hologram.helpers import StrEnum
 from dataclasses import dataclass
 from typing import Optional
+from thrift.transport.TSSLSocket import TSSLSocket
+import thrift
+import ssl
```

Comment:
Let's include these up in the `try` import (line 11) for requirements that may or may not be installed.

`ssl` is a net-new dependency, right? We'll need to add it in `setup.py`, within the `Pyhive` extra

Reply:
ssl is part of python library https://docs.python.org/3/library/ssl.html so i think it is not required to add

Reply:
Sorry, I said `ssl` here but I was thinking of `thrift_sasl`. That's a net-new dependency, right?